### PR TITLE
Add Sort and Complete task

### LINKED ISSUE #5: JS - Sorting by fields

### LINKED ISSUE #7: Change status when task completed

### PR DESCRIPTION
**Pull request by Linzeur**
- Close #5 
- Close #7

### CODE REVIEW

File: resources/js/index.js [hunk 1]
```diff
@@ -1,86 +1,163 @@
...
+    );
+  }
 }
 
-function validate(message, due_date) {
-  // message
-  if (message.value.trim().length == 0) {
-    message.focus();
-    return false;
+function sort_task(element, column) {
```
Comment (cdsandoval) [inline resources/js/index.js:132]: I like this function 👍

**Review by cdsandoval — APPROVED** [review]

File: resources/js/index.js [hunk 2]
```diff
@@ -1,86 +1,163 @@
...
   btnAddTask.addEventListener("click", save_task);
   let dateDue = document.getElementById("dateDue");
   // dateDue.value = new Date();
+  let columns = document.getElementsByClassName("colHeader");
+  let nColumns = columns.length;
+  for (let i = 0; i < nColumns; i++) {
+    let propertyName = columns[i].getAttribute("data-column");
+    columns[i].addEventListener(
+      "click",
+      sort_task.bind(null, columns[i], propertyName)
```
Comment (cdsandoval) [inline resources/js/index.js:39]: I like the use of bind here.

**Review by cdsandoval — APPROVED** [review]

File: resources/js/index.js [hunk 3]
```diff
@@ -47,3 +72,15 @@ function toggle_complete(element) {
   });
   sessionStorage.setItem("list", list_task);
 }
+
+function sort_task(this, column) {
+  if (list_task.length > 0) {
```
Comment (dotconde) [inline resources/js/index.js:144]: I agree 👁

File: resources/js/index.js [hunk 4]
```diff
@@ -47,3 +72,15 @@ function toggle_complete(element) {
...
+    let state = this.getAttribute("data-order");
+    list_task.sort(function(a, b) {
```
Comment (dotconde) [inline resources/js/index.js:147]: I like your sorting function

File: index.html [hunk 5]
```diff
@@ -55,11 +55,35 @@
...
-            <span>import</span>
-            <span>messages</span>
-            <span>due-date</span>
-            <span>creation-date</span>
-            <span>priority</span>
+            <span></span>
+            <span
+              class="colHeader"
+              data-column="message"
+              data-state="0"
```
Comment (dotconde) [inline index.html:62]: Could you please explain me how data-state works?

Comment (Linzeur) [reply index.html:62]: It is for indicating to us if the column is order by ascendent or descendent

**Pull request merged**